quinn-proto/src/endpoint.rs

```diff
@@ -188,17 +191,41 @@ impl Endpoint {
         //
 
         let addresses = FourTuple { remote, local_ip };
-        if let Some(ch) = self.index.get(&addresses, &first_decode) {
-            return Some(DatagramEvent::ConnectionEvent(
-                ch,
-                ConnectionEvent(ConnectionEventInner::Datagram {
-                    now,
-                    remote: addresses.remote,
-                    ecn,
-                    first_decode,
-                    remaining,
-                }),
-            ));
+        if let Some(route_to) = self.index.get(&addresses, &first_decode) {
+            let event = DatagramConnectionEvent {
+                now,
+                remote: addresses.remote,
+                ecn,
+                first_decode,
+                remaining,
+            };
+            match route_to {
+                RouteDatagramTo::Incoming(incoming_idx) => {
+                    let incoming_buffer = &mut self.incoming_buffers[incoming_idx];
+                    let server_config = self.server_config.as_ref().unwrap();
+                    let buffer_limit = (server_config.transport.receive_window.into_inner()
+                        as usize)
+                        .saturating_mul(2)
+                        .max(10_000);
```

Comment:
Why not use the receive window unmodified? If there's a strong justification, include it in a comment for posterity.

Reply:
There's some overhead, right? For packet and frame headers, crypto data, etc? Genuine question, tell me if I'm wrong. So I figured for safety: take the maximum amount of stream data allowed to be received, double it to be safe, and also lower-bound it at 10 KB for edge cases where the initial maximum stream data is configured so low that it's mostly overhead.

Reply:
There is some overhead, but it'll tend to be a single-digit percentage of the packet. At most I'd round up to the next packet size. If the sender wants to pack packets wastefully, then wasted space is ultimately on them, anyway.

Reply:
A clever way to round up to the next packet size without having to think too hard about what that *is* would be to drop a new packet iff the *existing* buffer is already above the limit.

Reply:
Done.